slamon_afm/models.py

```diff
@@ -7,10 +8,24 @@
     TypeDecorator
 from sqlalchemy.orm import relationship, backref
 from sqlalchemy.orm.exc import NoResultFound
-from sqlalchemy import func
+from sqlalchemy import func, event, case, literal_column
 
 db = SQLAlchemy()
 
+from slamon_afm.stats import statsd
+
+
+class TaskException(Exception):
+    task = None
+
+    def __init__(self, *args, task=None):
```

Comment:
the __str__ method of the Exception class will return all values passed to __init__.. The only motivation in the current solution is to pass readable message for the base Exeception, which could be also achieved by explicitly passing e.g. `msg` arg to the super.